clients/src/main/java/org/apache/kafka/common/requests/ApiVersionsRequest.java

```diff
@@ -51,23 +47,28 @@ public Builder(short version) {
 
         @Override
         public ApiVersionsRequest build(short version) {
-            return new ApiVersionsRequest(version);
+            return new ApiVersionsRequest(DATA, version);
         }
 
         @Override
         public String toString() {
-            return "(type=ApiVersionsRequest)";
+            return DATA.toString();
         }
     }
 
+    private static final Pattern SOFTWARE_NAME_VERSION_PATTERN = Pattern.compile("(?:[a-zA-Z0-9\\-.])+");
```

Comment:
Should we restrict the values for name and version? Maybe we can just test that they are non empty?


nit: the non-capture group isn't really necessary and this regex matches stuff like `"."` and `"---"`.

Reply:
We have discussed this in the KIP and the consensus is to have minimal validation (accepted characters) to avoid getting crap from the clients. It is especially important because name and version are used in metrics. 

Good point about the group. Thanks!

Reply:
Gotcha, makes sense. Should we maybe check that the first character is alphanumeric?

Reply:
Sure. I have updated the regex. It checks that the first and last character are alphanumeric.